### PR TITLE
Update Tooltip Component to use hooks and to be a FC

### PR DESCRIPTION
**Pull request by colinlohner**
## Overview 
Updated Tooltip Component to use hooks and to be a function component 

## Screenshots (if applicable) 
N/A

## Testing
- [x] Verify existing functionality remains as is

### CODE REVIEW

File: src/tooltip/src/Tooltip.js [hunk 1]
```diff
@@ -9,114 +9,64 @@ import TooltipStateless from './TooltipStateless'
...
-    showDelay: PropTypes.number.isRequired,
-
-    /**
-     * When True, manually show the Tooltip.
-     */
-    isShown: PropTypes.bool,
+const Tooltip = memo(props => {
+  const {
+    appearance = 'default',
+    position = Position.BOTTOM,
```
Comment (mshwery) [inline src/tooltip/src/Tooltip.js:15]: Any of these values that have defaults need to remove the `isRequired` from `propTypes` right?

File: src/tooltip/src/Tooltip.js [hunk 2]
```diff
@@ -9,114 +9,64 @@ import TooltipStateless from './TooltipStateless'
...
-    hideDelay: 120,
-    showDelay: 0
-  }
+  const handleHide = debounce(hide, hideDelay)
+
+  // Component will unmount
+  useEffect(
+    () => () => {
+      clearTimeout(closeTimeout)
+      setCloseTimeout(null)
```
Comment (mshwery) [inline src/tooltip/src/Tooltip.js:45]: No need to reset `setCloseTimeout`

File: src/tooltip/src/Tooltip.js [hunk 3]
```diff
@@ -161,69 +111,100 @@ export default class Tooltip extends PureComponent {
...
-            )}
-          >
-            {content}
-          </TooltipStateless>
-        )}
-      </Positioner>
-    )
+  // Tooltip was explicitly set to not be shown
+  if (propIsShown === false) {
+    shown = false
```
Comment (mshwery) [inline src/tooltip/src/Tooltip.js:125]: Couldn't this be combined with the above to make things a bit clearer?

```suggestion
  let shown = typeof propIsShown === 'boolean'
    ? propIsShown
    : (isShown || isShownByTarget) && !isPopoverShown()
```

File: src/tooltip/src/Tooltip.js [hunk 4]
```diff
@@ -161,69 +111,100 @@ export default class Tooltip extends PureComponent {
...
+      }}
+      isShown={shown}
+      position={position}
+      animationDuration={160}
+    >
+      {({ css, style, state, getRef }) => (
+        <TooltipStateless
+          id={id}
+          appearance={appearance}
+          ref={ref => getRef(ref)}
```
Comment (mshwery) [inline src/tooltip/src/Tooltip.js:141]: Can this just be `ref={getRef}`?

File: src/tooltip/src/Tooltip.js (hunk 1 above)
Comment (colinlohner) [reply src/tooltip/src/Tooltip.js:15]: Yes. My bad

File: src/tooltip/src/Tooltip.js [hunk 5]
```diff
@@ -118,12 +117,9 @@ const Tooltip = memo(props => {
     setIsShownByTarget(true)
   }
 
-  let shown = (propIsShown || isShown || isShownByTarget) && !isPopoverShown()
-
-  // Tooltip was explicitly set to not be shown
-  if (propIsShown === false) {
-    shown = false
-  }
```
Comment (mshwery) [inline src/tooltip/src/Tooltip.js]: Actually now that I'm looking at this again I think this is meant to only capture when `isShown` is explicitly false. When it's set to `true` we do indeed need `&& !isPopoverShown()` as well.

Maybe revert my suggestion 😬

Comment (colinlohner) [reply src/tooltip/src/Tooltip.js]: lol okay

**Pull request merged**